src/main/java/com/g4s8/ghman/web/PsNumericFlag.java

```diff
@@ -0,0 +1,31 @@
+package com.g4s8.ghman.web;
+
+import java.io.IOException;
+import org.takes.Response;
+import org.takes.facets.auth.Identity;
+import org.takes.facets.auth.Pass;
+import org.takes.misc.Opt;
+import org.takes.rq.RqHref;
+import org.takes.Request;
+
+/**
+ * Numeric Flag identity extraction
+ * @since 1.0
+ * @checkstyle LineLengthCheck (500 lines)
+ * @checkstyle ClassDataAbstractionCouplingCheck (500 lines)
+ * @checkstyle ClassFanOutComplexityCheck (500 lines)
+ */
+final class PsNumericFlag implements Pass {
+
+    @Override
+    public Opt<Identity> enter(final Request req) throws IOException {
+        return new Opt.Single<>(
+            new Identity.Simple(String.format("urn:uid:%s", new RqHref.Smart(req).single("ps"))));
```

Comment:
@samihibrahim please replace the call to static method `String.format` with the use of cactoos' `text.Formatted` class.